exo/networking/manual/manual_discovery.py

```diff
@@ -69,3 +68,45 @@ async def task_find_peers_from_config(self):
       await asyncio.sleep(1.0)
 
       if DEBUG_DISCOVERY >= 2: print(f"Current known peers: {[peer.id() for peer in self.known_peers.values()]}")
+
+  async def task_clean_up_peers_from_config(self):
+    if DEBUG_DISCOVERY >= 2: print("Starting task to clean up peers from config...")
+    while True:
+      peers_from_config = self._get_peers()
+      if peers_from_config:
+        peers_to_remove = [peer for peer in self.known_peers.keys() if peer not in peers_from_config]
+
+        for peer in peers_to_remove:
+          if DEBUG_DISCOVERY >= 2: print(f"{peer} is no longer found in the config but is currently a known peer. Removing from known peers...")
+          try: del self.known_peers[peer]
+          except KeyError: pass
+
+      await asyncio.sleep(1.0)
+
+  def _get_peers(self):
+    try:
+      current_mtime = os.path.getmtime(self.network_config_path)
+
+      if self._cached_peers is not None and self._last_modified_time is not None and current_mtime <= self._last_modified_time:
+        return self._cached_peers
+
+      topology = NetworkTopology.from_path(self.network_config_path)
```

Comment:
Is `from_path` blocking? Same comment as above. Can we make all I/O operations to `self.network_config_path` happen on a thread pool executor with one thread?